Assets/_BForBoss/_Core/Scripts/Boss/DerekHealthBehaviour.cs

```diff
@@ -1,70 +1,87 @@
 using System;
+using Perigon.Utility;
 using Perigon.Weapons;
 using UnityEngine;
 
 namespace BForBoss
 {
     public class DerekHealthBehaviour : MonoBehaviour, IBulletCollision
     {
-        [SerializeField] private int _maxHealth = 12;
+        [SerializeField] private int _maxHealth = 20;
         [SerializeField] private int _damagePerShot = 1;
 
+        [SerializeField] private DerekHealthView _healthView;
+
         private int _elapsedHealth;
         private float _threshold;
         private DerekContextManager.Vulnerability _vulnerabilityState = DerekContextManager.Vulnerability.Invulnerable;
-        
+
+        private bool _canReceiveDamage = false;
+
         private event Action<DerekHealthViewState> OnStateChanged;
         private event Action OnThresholdHit;
 
-        public DerekContextManager.Vulnerability VulnerabilityState
+        public bool CanReceiveDamage
```

Comment:
instead of `CanReceiveDamage` should stick with the naming keyword of `vulnerbility`. 

If not, it'll be confusing, you have vulnerbility vs receiveDamage.
and then you have `!canReceiveDamage` with the negation which is even more harder to read.